### DISCUSSION TITLE
Are "tag" and "node name" the same?

### DISCUSSION THREAD

**Discussion by jihchi** [kdl-org/kdl#238] [Q&A] [answered]
In the [KDL Query Language Spec](https://github.com/kdl-org/kdl/blob/main/QUERY-SPEC.md#matchers), I see there are a few matchers that seems referring the same thing:

1. `(foo)`: Selects any element with a tag named `foo`
1. `[name() = "hi"]`: Selects any element whose *node name* is "hi". Equivalent to just `hi`, but more useful when using string operators
1. `[tag() = "hi"]`: Selects any element whose tag is "hi". Equivalent to just `(hi)`, but more useful when using string operators

Do I understand correctly?

**Comment by zkat** [comment, marked as answer]
"tag" is short for "type tag", as in the "foo" in (foo)mynode.

"node name" is the name of the node itself. The "mynode" above.

**Reply by larsgw** [reply]
I can update the documentation to remove usage of "tag", I think that was my fault as we called it type annotation before.

**Comment by jihchi** [comment]
I also found that there is an example in the [comment section](https://github.com/kdl-org/kdl#comments):

```
// C style

/*
C style multiline
*/

tag /*foo=true*/ bar=false

/*/*
hello
*/*/
```

where `tag /*foo=true*/ bar=false` just happen to be a node named `tag`?

**Reply by larsgw** [reply]
Yes.

**Comment by jihchi** [comment]
Thank you @zkat , @larsgw
